test/clojure-ts-mode-util-test.el

```diff
@@ -23,7 +23,110 @@
+    (with-clojure-ts-buffer "(ns
+    foo)"
+                            (expect (clojure-ts-find-ns) :to-equal "foo"))
+    (with-clojure-ts-buffer "(ns foo.baz)"
+                            (expect (clojure-ts-find-ns) :to-equal "foo.baz"))
+    (with-clojure-ts-buffer "(ns ^:bar foo)"
+                            (expect (clojure-ts-find-ns) :to-equal "foo"))
+    (with-clojure-ts-buffer "(ns ^:bar ^:baz foo)"
+                            (expect (clojure-ts-find-ns) :to-equal "foo")))
+
+  (it "should find namespaces with spaces before ns form"
+    (with-clojure-ts-buffer "  (ns foo)"
+                            (expect (clojure-ts-find-ns) :to-equal "foo")))
+
+  (it "should skip namespaces within any comment forms"
+    (with-clojure-ts-buffer "(comment
+      (ns foo))"
+                            (expect (clojure-ts-find-ns) :to-equal nil))
+    (with-clojure-ts-buffer " (ns foo)
+     (comment
+      (ns bar))"
+                            (expect (clojure-ts-find-ns) :to-equal "foo"))
+    (with-clojure-ts-buffer " (comment
+      (ns foo))
+     (ns bar)
+    (comment
+      (ns baz))"
+                            (expect (clojure-ts-find-ns) :to-equal "bar")))
+
+  (it "should find namespace declarations with nested metadata and docstrings"
+    (with-clojure-ts-buffer "(ns ^{:bar true} foo)"
+                            (expect (clojure-ts-find-ns) :to-equal "foo"))
+    (with-clojure-ts-buffer "(ns #^{:bar true} foo)"
+                            (expect (clojure-ts-find-ns) :to-equal "foo"))
+    (with-clojure-ts-buffer "(ns #^{:fail {}} foo)"
+                            (expect (clojure-ts-find-ns) :to-equal "foo"))
+    (with-clojure-ts-buffer "(ns ^{:fail2 {}} foo.baz)"
+                            (expect (clojure-ts-find-ns) :to-equal "foo.baz"))
+    (with-clojure-ts-buffer "(ns ^{} foo)"
+                            (expect (clojure-ts-find-ns) :to-equal "foo"))
+    (with-clojure-ts-buffer "(ns ^{:skip-wiki true}
+      aleph.netty)"
+                            (expect (clojure-ts-find-ns) :to-equal "aleph.netty"))
+    (with-clojure-ts-buffer "(ns ^{:foo {:bar :baz} :fake (ns in.meta)} foo
+  \"docstring
+(ns misleading)\")"
+                            (expect (clojure-ts-find-ns) :to-equal "foo")))
+
+  (it "should support non-alphanumeric characters"
+    (with-clojure-ts-buffer "(ns foo+)"
+                            (expect (clojure-ts-find-ns) :to-equal "foo+"))
+    (with-clojure-ts-buffer "(ns bar**baz$-_quux)"
+                            (expect (clojure-ts-find-ns) :to-equal "bar**baz$-_quux"))
+    (with-clojure-ts-buffer "(ns aoc-2019.puzzles.day14)"
+                            (expect (clojure-ts-find-ns) :to-equal "aoc-2019.puzzles.day14")))
+
+  (it "should support in-ns forms"
+    (with-clojure-ts-buffer "(in-ns 'bar.baz)"
+                            (expect (clojure-ts-find-ns) :to-equal "bar.baz")))
+
+  (it "should take the first ns instead of closest unlike clojure-mode"
```

Comment:
This is the only scenario where it differs clojure-find-ns after this PR.

Reply:
Is there any particular reason for this change?

Reply:
AFAIU It's debatable what is the correct behaviour here.
clojure-mode finds the closest ns above point and falls back to finding the closest ns below point.

1) My understanding is that clojure-mode finding the closest ns is more of a result of an implementation detail rather than desired behaviour. 

2)Also I haven't figured out yet how to find all occurrences or do a reverse search with tree-sitter queries.

Reply:
I recall in the beginning `clojure-mode` was just looking for a `ns` form in the beginning of the file, but then we changed it, as per the Clojure specification the current `ns` was supposed to be whatever ns is closest above the form in question. I don't recall doing a search below as well, but perhaps there was some reason for this.

Reply:
Anyways, if by first you mean "closest above" (and so it seems looking at the tests), I think that's fine.

Reply:
Right now it's first from the beginning of file. (point is not used at all). 
This is the behaviour before the PR as well. I haven't updated it as it would require more complicated tree-sitter queries.

The below test is matching the 3rd ns only because the first 2 are invalid not because of the point.

Reply:
If clojure-mode behaviour is ideal I would prefer to create a new issue and handle it in a separate PR.(Need more tree-sitter knowledge to handle the queries there.)

Reply:
That'd be fine by me.

Reply:
In general that has never been super important for me (to do this by the letter), as usually source files have only the leading `ns` form. But clearly it was a problem for some users of `clojure-mode`. :D

Reply:
@p4v4n If you'd like to chat about tree-sitter-clojure, there is a gitter / matrix channel.

See the bottom of [this](https://github.com/sogaiu/tree-sitter-clojure/issues/33#issue-1503922369) for links if interested.